test/Microsoft.AspNet.Mvc.Core.Test/LocalRedirectResultTest.cs

```diff
@@ -0,0 +1,137 @@
+            var actionContext = GetActionContext(httpContext);
+            var result = new LocalRedirectResult(contentPath);
+
+            // Act
+            result.ExecuteResult(actionContext);
+
+            // Assert
+            httpResponse.Verify();
+        }
+
+        [Theory]
+        [InlineData("", "Home/About", "/Home/About")]
+        [InlineData("/myapproot", "http://www.google.com", "/test")]
+        public void Execute_Throws_ForNonLocalUrl(
+            string appRoot,
+            string contentPath,
+            string expectedPath)
+        {
+            // Arrange
+            var httpResponse = new Mock<HttpResponse>();
+            httpResponse.Setup(o => o.Redirect(expectedPath, false))
+                        .Verifiable();
+
+            var httpContext = GetHttpContext(appRoot, contentPath, expectedPath, httpResponse.Object);
+            var actionContext = GetActionContext(httpContext);
+            var result = new LocalRedirectResult(contentPath);
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(() => result.ExecuteResult(actionContext));
+            Assert.Equal("The supplied Url is not local.", exception.Message);
+        }
+
+        private static ActionContext GetActionContext(HttpContext httpContext)
+        {
+            var routeData = new RouteData();
+            routeData.Routers.Add(new Mock<IRouter>().Object);
+
+            return new ActionContext(httpContext,
+                                    routeData,
+                                    new ActionDescriptor());
+        }
+
+        private static IServiceProvider GetServiceProvider(IUrlHelper urlHelper)
+        {
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddInstance<IUrlHelper>(urlHelper);
+            return serviceCollection.BuildServiceProvider();
+        }
+
+        private static HttpContext GetHttpContext(
+            string appRoot,
+            string contentPath,
+            string expectedPath,
+            HttpResponse response)
+        {
+            var httpContext = new Mock<HttpContext>();
+            var actionContext = GetActionContext(httpContext.Object);
+            var actionContextAccessor = new ActionContextAccessor() { ActionContext = actionContext };
+            var mockActionSelector = new Mock<IActionSelector>();
+            var urlHelper = new UrlHelper(actionContextAccessor, mockActionSelector.Object);
+            var serviceProvider = GetServiceProvider(urlHelper);
+
+            httpContext.Setup(o => o.Response)
+                       .Returns(response);
+            httpContext.SetupGet(o => o.RequestServices)
+                       .Returns(serviceProvider);
+            httpContext.Setup(o => o.Request.PathBase)
+                       .Returns(new PathString(appRoot));
+
+            return httpContext.Object;
```

Comment:
Can you avoid using mocks for this kind of trivial stuff? I think the setup code using `ServiceCollection` and `DefaultHttpContext` will be much more concise.